src/js/globals.d.ts

```diff
@@ -208,3 +216,23 @@ declare module "worker-loader?inline=true&fallback=false!*" {
 
     export default WebpackWorker;
 }
+
+// JSX
+declare namespace JSX {
+    // The return type of our JSX Factory: this could be anything
+    type Element = HTMLElement;
+
+    // IntrinsicElementMap grabs all the standard HTML tags in the TS DOM lib.
+    interface IntrinsicElements extends IntrinsicElementMap {}
+
+    // The following are custom types, not part of TS's known JSX namespace:
+    type IntrinsicElementMap = {
+        [K in keyof HTMLElementTagNameMap]: {
+            [k: string]: any;
+        };
+    };
+
+    interface Component {
+        (properties?: { [key: string]: any }, children?: Node[]): Node;
+    }
+}
```

Comment:
this appears to have been taken from <https://stackoverflow.com/a/68238924> without credit!!! (and it's CC BY-SA 4.0 too!!!!!!!)

Reply:
it is compatible though, yay https://creativecommons.org/share-your-work/licensing-considerations/compatible-licenses

Reply:
credited